contribute/style-guides/e2e.md

```diff
@@ -0,0 +1,140 @@
+
+The next step is to create a `Page` representation in our e2e test framework to glue the test with the real implementation using the `pageFactory` function. For that function we can supply an `url` and `selectors` like the example below.
+```typescript
+export const Login = pageFactory({
+  url: '/login', // used when called from Login.visit()
+  selectors: {
+    username: 'Username input field', // used when called from Login.username().type('Hello World')
+  },
+});
+```
+
+The next step is to add `Login` page to the exported const `Pages` in `packages/grafana-e2e/src/pages/index.ts` so that it appears when we type `e2e.pages` in our IDE.
+```ecmascript 6
+export const Pages = {
+  Login,
+  ...,
+  ...,
+  ...,
+};
+
+```
+Now that we have a `Page` called `Login` in our `Pages` const we can use that to add a selector in our html like shown below and now this really signalizes to future developers that this is part of a e2e test.   
+```jsx harmony
+<div>
+    <input type="text" className="gf-form-input login-form-input" aria-label={e2e.pages.Login.selectors.username}/>
+</div>
+```
+
+The last step in our example is to use our `Login` page as part of a test. The `pageFactory` function we used before gives us 2 things by using it. 
+- Firstly the `url` property will be used whenever we call the `visit` function and is equivalent to the Cypress function [cy.visit()](https://docs.cypress.io/api/commands/visit.html#Syntax)
+- Secondly any defined selector in the `selectors` property can be accessed from `Login` page by invoking it and this is equivalent to the result of the Cypress function [cy.get(...)](https://docs.cypress.io/api/commands/get.html#Syntax) 
+```ecmascript 6
+describe('Login test', () => {
+  it('Should pass', () => {
+    e2e.pages.Login.visit();
+    e2e.pages.Login.username().type('admin');
+  });
+});
+```
+
+## Advanced example
+Lets take a look at an example that uses the same `selector` for multiple items in a list for instance. In this example app we have an list of data sources that we during a e2e test want to click on.
+
+```jsx harmony
+<ul>
+  {dataSources.map(dataSource => (
+    <li className="card-item-wrapper" key={dataSource.id}>
+      <a className="card-item" href={`datasources/edit/${dataSource.id}`}>
+        <div className="card-item-name">
+          {dataSource.name}
+        </div>
+      </a>
+    </li>
+  ))}
+</ul>
+```
+```
+
+Just as before in the basic example we'll start by creating a page abstraction using the `pageFactory` function. 
+```typescript
+export const DataSources = pageFactory({
+  url: '/datasources',
+  selectors: {
+    dataSources: (dataSourceName: string) => `Data source list item ${dataSourceName}`,
+  },
+});
+```
+You might have noticed that instead of a simple `string` as the `selector` we're using a `function` that takes a string parameter as an argument and returns a formatted string using the argument.
```

Comment:
```suggestion
You might have noticed that instead of a simple `string` as the `selector`, we're using a `function` that takes a string parameter as an argument and returns a formatted string using the argument.
```